File: go/02_cryptographic_failures.go
// A02 - Cryptographic Failures. VULN: MD5 + hardcoded AES key.
package main

import (
	"crypto/md5"
	"encoding/hex"
)

var aesKey = []byte("sixteen-byte-key") // VULN: A02

func HashPassword(pw string) string {
	h := md5.Sum([]byte(pw))
	return hex.EncodeToString(h[:])
}

File: javascript/02_cryptographic_failures.js
// A02 - Cryptographic Failures. VULN: SHA1 for password, static pepper in source.
const crypto = require("crypto");

const PEPPER = "static-pepper-123"; // VULN: A02

function hashPassword(password) {
  return crypto.createHash("sha1").update(password + PEPPER).digest("hex");
}

module.exports = { hashPassword };
